app/uk/gov/hmrc/economiccrimelevyregistration/services/EclRegistrationService.scala

```diff
@@ -48,4 +50,79 @@ class EclRegistrationService @Inject() (
 
   def upsertRegistration(registration: Registration)(implicit hc: HeaderCarrier): Future[Registration] =
     eclRegistrationConnector.upsertRegistration(registration)
+
+  def getSubscription(eclRegistration: String)(implicit hc: HeaderCarrier): Future[GetSubscriptionResponse] =
+    eclRegistrationConnector.getSubscription(eclRegistration)
+
+  def transformToRegistration(
+    registration: Registration,
+    getSubscriptionResponse: GetSubscriptionResponse
+  ): Registration = {
+    val primaryContact      = getSubscriptionResponse.primaryContactDetails
+    val secondaryContact    = getSubscriptionResponse.secondaryContactDetails
+    val subscriptionAddress = getSubscriptionResponse.correspondenceAddressDetails
+
+    val firstContactDetails: ContactDetails = ContactDetails(
+      Some(primaryContact.name),
+      Some(primaryContact.positionInCompany),
+      Some(primaryContact.emailAddress),
+      Some(primaryContact.telephone)
+    )
+    val secondContactDetails                = secondaryContact match {
+      case Some(value) =>
+        ContactDetails(
+          Some(value.name),
+          Some(value.positionInCompany),
+          Some(value.emailAddress),
+          Some(value.telephone)
+        )
+      case _           => ContactDetails.empty
+    }
+    val secondContactPresent                = if (secondaryContact.isEmpty) {
```

Comment:
this doesn't have to be wrapped in an if block

`val secondContactPresent = Some(secondaryContact.isDefined)`

Reply:
Done